walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,80 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    int minutesInOneDay = 24 * 60;
+    if (request.getDuration() > minutesInOneDay)
+        return new ArrayList<TimeRange>();
+
+    Collection<TimeRange> withOptional = new ArrayList<>();
+    Collection<TimeRange> mandatoryOnly = new ArrayList<>();
```

Comment:
I'm not sure if it matters but should there be a `TimeRange` in between the `<>`?

Reply:
When I coded in intelliJ for school it would get mad if I put the type in the constructor (no clue why, I think it said it was redundant or something), so it became habit to just leave it out. I think it infers it from typing the variable